scipy/stats/tests/test_multivariate.py

```diff
@@ -2465,6 +2466,160 @@ def test_invalid_params(self):
+            random_table([1, 2], [1, 0])
+
+        row = [1, 3]
+        col = [2, 1, 1]
+        r, c, n = random_table._process_parameters([1, 3], [2, 1, 1])
+        assert_equal(row, r)
+        assert_equal(col, c)
+        assert n == np.sum(row)
+
+    def test_process_rvs_method_on_None(self):
+        # TODO also test with a case where "patefield" is selected
+        row = [1, 3]
+        col = [2, 1, 1]
+
+        ct = random_table
+        expected = ct.rvs(row, col, method="boyett", random_state=1)
+        got = ct.rvs(row, col, method=None, random_state=1)
+
+        assert_equal(expected, got)
+
+    def test_rvs_method(self):
+        pytest.xfail("patefield not yet implemented")
+
+        row = [1, 3]
+        col = [2, 1, 1]
+
+        ct = random_table
+        rvs1 = ct.rvs(row, col, size=1000, method="boyett", random_state=1)
+        rvs2 = ct.rvs(row, col, size=1000, method="patefield", random_state=1)
+
+        tab1, count1 = np.unique(rvs1, axis=0, return_counts=True)
+        tab2, count2 = np.unique(rvs2, axis=0, return_counts=True)
+
+        assert_equal(tab1, tab2)
+        assert_allclose(count1, count2)
+
+    def test_process_rvs_method_bad_argument(self):
+        row = [1, 3]
+        col = [2, 1, 1]
+
+        # order of items in set is random, so cannot check that
+        message = "'foo' not recognized, must be one of"
+        with pytest.raises(ValueError, match=message):
+            random_table.rvs(row, col, method="foo")
+
+    def test_pmf(self):
+        row = [2, 6]
+        col = [1, 3, 4]
+        rvs = random_table.rvs(row, col, size=1000, random_state=123)
+
+        unique_rvs, counts = np.unique(rvs, axis=0, return_counts=True)
+
+        p = random_table.pmf(unique_rvs)
+        p0 = random_table.pmf(unique_rvs[0])
+        assert_equal(p0, p[0])
+        assert_allclose(p * len(rvs), counts, rtol=0.1)
+
+        # accept any iterable
+        p0_2 = random_table.pmf(list(unique_rvs[0]))
+        assert_equal(p0, p0_2)
+
+        # accept high-dimensional input
+        rng = np.random.default_rng(1)
+        z = rng.integers(4, 9, size=(2, 3, 4, 5))
+        p = random_table.pmf(z)
+        assert p.shape == (2, 3)
+        for i in range(p.shape[0]):
+            for j in range(p.shape[1]):
+                pij = p[i, j]
+                zij = z[i, j]
+                qij = random_table.pmf(zij)
+                assert_equal(pij, qij)
+
+    def test_logpmf(self):
+        row = [2, 6]
+        col = [1, 3, 4]
+        rvs = random_table.rvs(row, col, size=1000, random_state=123)
+
+        unique_rvs, counts = np.unique(rvs, axis=0, return_counts=True)
+
+        lp0 = random_table.logpmf(unique_rvs[0])
+        lp = random_table.logpmf(unique_rvs)
+        assert_equal(lp0, lp[0])
+        assert_allclose(np.exp(lp) * len(rvs), counts, rtol=0.1)
+
+        # accept any iterable
+        lp0_2 = random_table.logpmf(list(unique_rvs[0]))
+        assert_equal(lp0, lp0_2)
```

Comment:
Even though it doesn't matter for `assert_equal`, I should mention that the argument order is [supposed to be](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html) `actual, desired`. It matters for `assert_allclose` with relative tolerances (and it's correct above). Here, I'd consider `lp0` to be reference value.

Reply:
Ok, good point. This is covered by your patch, so nothing for me to do.